### DISCUSSION TITLE
Using the pointset structuring with region growing surface detection results

### DISCUSSION THREAD

**Discussion by Sushant-Chavan** [CGAL/cgal#5954] [Q&A] [answered]
Hi,
Thank you for providing this wonderful library!

I am new to the CGAL library and am trying to use the [point set structuring](https://doc.cgal.org/latest/Point_set_processing_3/index.html#Point_set_processing_3Structuring) algorithm on a pointset. However, unlike the [example](https://doc.cgal.org/latest/Point_set_processing_3/index.html#Point_set_processing_3Example_9) which uses the [Efficient RANSAC](https://doc.cgal.org/latest/Shape_detection/index.html#Shape_detection_RANSAC) shape detection method, I want to use the [Region Growing surface detection](https://doc.cgal.org/latest/Shape_detection/index.html#Shape_detection_RegionGrowing) method to detect planes prior to applying the structuring algorithm. 

However, I am unable to find the right way to do this. While the Efficient RANSAC shape detection provides a convinience method to access the detected planes (as used in the example), the region growing algorithm does not have any such methods. 

Can someone please show me an example code on how the planes detected by the Region Growing surface detection algorithm be used as the input the point set structuring algorithm?

Thanks!
Sushant

**Comment by danston** [comment, marked as answer]
Hi @Sushant-Chavan It is a nice suggestion to add such or a similar example. What you should do is the following:

- Run region growing on your point set.
- You get indices of all points, which form planar regions.
- Access these points and apply [least_squares_fitting_3()](https://doc.cgal.org/latest/Principal_component_analysis/group__PkgPrincipalComponentAnalysisDLLSF3.html#gadcf556c115c6bac30a54bf7fdd0d32e3) function that returns a plane for each region.
- Then provide these planes to the structuring algorithm through [this function](https://doc.cgal.org/latest/Point_set_processing_3/group__PkgPointSetProcessing3Algorithms.html#gaebbc04909dfd575a94f932c8d3bacdce) together with points that you used for fitting.
